test/earray.c

```diff
@@ -2303,7 +2303,7 @@ main(void)
     h5_test_init();
     fapl = h5_fileaccess();
     if (TestExpress > 1)
-        printf("***Express test mode on.  Some tests may be skipped\n");
+        printf("***Express test mode %d.  Some tests may be skipped\n", TestExpress);
```

Comment:
Not sure what you want here, since the variable is just an int?

Reply:
Cast LocalTestExpress to an int:  "LocalTestExpress" => "(int)LocalTestExpress"

Reply:
Seems like a bad idea to cast something that doesn't really need casting since it'll just hide future possible errors.

Reply:
If the type of the variable ever changes it can produce memory errors. Modern compilers might be able to detect and throw a warning in this case (if the type doesn't match the format string), I'm not sure.

Reply:
I think you want the compiler to report the issue and not cover it up?

Reply:
I'm not certain the compiler will report an issue if there's no cast. And it is much more unsafe to pass the wrong type to printf than to cast something to the right type - the only potential problem there is wrap around. printf looks at the format string and reads that many bytes from the input parameters, it does  not check if enough data was passed in the parameters, so if the types don't match exactly it's possible for printf to read past the end of the input parameters which is a memory error.

Reply:
It looks like gcc does not warn by default but does if -Wformat is on

Reply:
I believe we have Wformat on by default for GCC and most of the other compilers. My main concern is that casting to int can hide other problems if the variable ever does change to something else, but it seems unlikely that it will at this point. Absolutely not an important issue though; either way is fine.